src/webserver/routes/api/currency.ts

```diff
@@ -430,9 +436,16 @@ router.delete(
 				return res
 					.status(403)
 					.send('You are not allowed to delete this transaction.');
-
-			transaction.deleted = true;
-			await transaction.save();
+			const storeReq = await StoreRequest.findOne({
+				transactionID: transaction._id,
+			});
+			if (storeReq) {
+				await storeReq.delete();
+				await transaction.delete();
+			} else {
+				transaction.deleted = true;
+				await transaction.save();
+			}
```

Comment:
I can understand deleting the store request, but why delete the transaction itself two different ways?
One marks the transaction as deleted and the other removes it from existence. Shouldn't we just always mark it as deleted, like we do in every other case?

Reply:
I think I wanted to distinguish between deleted store request transactions (where a request was never approved and no money ever actually changed hands) as opposed to a normal transaction (where money has changed hands). We can treat all the same though if you think that makes more sense?